esrally/telemetry.py

```diff
@@ -568,6 +568,138 @@ def record(self):
                 self.metrics_store.put_doc(doc, level=MetaInfoScope.cluster, meta_data=shard_metadata)
 
 
+class ShardStats(TelemetryDevice):
+    """
+    Collects and pushes shard stats for the specified cluster to the metric store.
+    """
+    internal = False
+    command = "shard-stats"
+    human_name = "Shard Stats"
+    help = "Regularly samples nodes stats at shard level"
+
+    def __init__(self, telemetry_params, clients, metrics_store):
+        """
+        :param telemetry_params: The configuration object for telemetry_params.
+            May optionally specify:
+            ``shard-stats-indices``: JSON structure specifying the index pattern per cluster to publish stats from.
+            Not all clusters need to be specified, but any name used must be be present in target.hosts. Alternatively,
+            the index pattern can be specified as a string can be specified in case only one cluster is involved.
+            Example:
+            {"shard-stats-indices": {"cluster_a": ["follower"],"default": ["leader"]}
+
+            ``shard-stats-sample-interval``: positive integer controlling the sampling interval. Default: 60 seconds.
+        :param clients: A dict of clients to all clusters.
+        :param metrics_store: The configured metrics store we write to.
+        """
+        super().__init__()
+
+        self.telemetry_params = telemetry_params
+        self.clients = clients
+        self.sample_interval = telemetry_params.get("shard-stats-sample-interval", 60)
+        if self.sample_interval <= 0:
+            raise exceptions.SystemSetupError(
+                f"The telemetry parameter 'shard-stats-sample-interval' must be greater than zero but was {self.sample_interval}.")
+
+        self.specified_cluster_names = self.clients.keys()
+        indices_per_cluster = self.telemetry_params.get("shard-stats-indices", False)
+        # allow the user to specify either an index pattern as string or as a JSON object
+        if isinstance(indices_per_cluster, str):
+            self.indices_per_cluster = {opts.TargetHosts.DEFAULT: indices_per_cluster}
+        else:
+            self.indices_per_cluster = indices_per_cluster
+
+        if self.indices_per_cluster:
+            for cluster_name in self.indices_per_cluster.keys():
+                if cluster_name not in clients:
+                    raise exceptions.SystemSetupError(
+                        f"The telemetry parameter 'shard-stats-transforms' must be a JSON Object with keys "
```

Comment:
The parameter is called `shard-stats-indices`, can you please align the names? Also, we should add user docs.